test_runner/src/main/kotlin/ftl/args/CreateAndroidArgs.kt

```diff
@@ -42,5 +45,19 @@ fun createAndroidArgs(
     obbFiles = gcloud::obbfiles.require(),
     obbNames = gcloud::obbnames.require(),
     grantPermissions = gcloud.grantPermissions,
-    testTargetsForShard = gcloud.testTargetsForShard?.normalizeToTestTargets().orEmpty()
+    testTargetsForShard = gcloud.testTargetsForShard?.normalizeToTestTargets().orEmpty(),
+    customSharding = createCustomShards(commonArgs.shardingJson)
 )
+
+private fun createCustomShards(shardingJsonPath: String?) =
+    if (shardingJsonPath.isNullOrBlank()) emptyMap()
+    else {
+        fromJson<Map<String, AndroidTestShards>>(
+            Paths.get(shardingJsonPath).toFile().readText()
+        ).mapValues { (_, shards) ->
+            shards.copy(
+                app = shards.app.normalizeFilePath(),
+                test = shards.test.normalizeFilePath()
+            )
+        }
```

Comment:
move to separate function please

Reply:
hm...it's already a standalone function but I can try to make it simpler 👍🏻